prog-o-meter.py

```diff
@@ -1,390 +1,419 @@
+        """
+        # Attributes
+        self.root = Tk.Tk()
+        self.root.title("Login")
+        self.username = ""
+        self.user_type = _user_type
+        # Tkinter instantiation
+        self.canvas_layout()
+        self.input_button()
+        self.root.protocol("WM_DELETE_WINDOW", lambda: quit())
+        self.root.mainloop()
+    def canvas_layout(self):
+        """Display a Tkinter canvas.
+
+        Creates a 300x50 px canvas with a greeting text and an entry widget (input field for text).
+
+        Attributes:
+            CANVAS_WIDTH: The width of the canvas is hardcoded to 300 px.
+            CANVAS_HEIGHT: The height of the canvas is hardcoded to 50 px.
+            canvas: The Tkinter canvas widget.
+            text_entry: A Tkinter text-entry widget (input field for text)
+        """
+        self.CANVAS_WIDTH = 300
+        self.CANVAS_HEIGHT = 50
+        self.canvas = Tk.Canvas(self.root, width = self.CANVAS_WIDTH, height = self.CANVAS_HEIGHT)
+        self.canvas.pack()
+        self.text_entry = Tk.Entry(self.root)
+        self.text_entry.pack()
+        self.text_entry.focus_force()
+        if self.user_type == 1:        # Display appropriate greeting for returning users
+            self.canvas.create_text(self.CANVAS_WIDTH/2, 20, text = "Good to see you again! Please enter your name")
+        elif self.user_type == 2:        # Display appropriate greeting for new users
+            self.canvas.create_text(self.CANVAS_WIDTH/2, 20, text = "Lets get you started! Please enter your name")
+    def input_button(self):
+        """Display the inout button on the canvas.
+
+        Displays a submit button, for user to click when they have typed in their name.
+        When button is clicked, it stores the input name as username, and then closes the Tkinter window.
+
+        Attributes:
+            submit_button: Button with the text "Submit", which calls the function save_and_close
+        """
+        self.submit_button = Tk.Button(self.root, text = "Submit", command = self.save_and_close)
+        self.return_button = Tk.Button(self.root, text = "Go back to start", command = lambda: return_to_start(self.root))
+        self.return_button.pack(side='left', expand=True, pady=10)
+        self.submit_button.pack(side='left', expand=True, pady=10)
+        self.root.bind('<Return>', self.save_and_close)
+
+    def save_and_close(self, event=None):
+        """Save input text as username, then close Tkinter window.
+
+        If user is a returning user, first checks to see whether the entered text is an existing username, then displays a warning if
+        it is not. User can then enter a new name.
+        """
+        if not os.path.isfile(self.text_entry.get()+'.txt') and self.user_type == 1:    #if user is a returning user, check if there's a file with the given name
```

Comment:
I think it would make sense to flip these two statements, i.e. check if it is a returning user before we check if the file exists. Just because if it is not a returning user, it seems like a lot to go through checking if the file exists. But maybe I've overlooked some good reason for doing it the other way around.